eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py

```diff
@@ -100,19 +102,62 @@ def fill_in_from_previous_inventory(private_config_xml_file, release_version):
             execute_query(pg_conn, query_insert)
 
 
+def fill_num_rs_id_for_taxonomy_and_assembly(mongo_source, private_config_xml_file):
+    group_rs_by_taxonomy_and_assembly = {
+        "$group":
+            {
+                "_id": {"taxonomy": "$tax", "assembly": "$asm"},
+                "rs_count": {"$sum": 1},
+            }
+    }
+    cve_collection = mongo_source.mongo_handle[mongo_source.db_name]["clusteredVariantEntity"]
+    aggregated_rs_info = cve_collection.aggregate([group_rs_by_taxonomy_and_assembly], batchSize=0)
+    for agg_info in aggregated_rs_info:
+        with get_metadata_connection_handle("development", private_config_xml_file) as pg_conn:
+            taxonomy = agg_info["_id"]["taxonomy"]
+            assembly = agg_info["_id"]["assembly"]
+            rs_count = agg_info["rs_count"]
+            logger.info(f'updating rs count({rs_count}) for taxonomy({taxonomy}) and assembly({assembly})')
+            query_update = (
+                f'UPDATE eva_progress_tracker.clustering_release_tracker SET num_rs_to_release={rs_count} '
+                f'WHERE taxonomy={taxonomy} and assembly_accession=\'{assembly}\'')
```

Comment:
In some case, the row won't exists already so this update won't work. You'll have to create the whole row in that case.

Reply:
done